src/psyclone/psyir/nodes/dynamic_omp_task_directive.py

```diff
@@ -0,0 +1,1998 @@
+                "parent_node"
+            ].copy()
+            for temp_ref in self._proxy_loop_vars[index_symbol]["parent_node"]:
+                real_ref = temp_ref.copy()
+                # We have a Literal step value, and a Literal in
+                # the Binary Operation. These Literals must both be
+                # Integer types, so we will convert them to integers
+                # and do some divison.
+                step_val = int(parent_loop.step_expr.value)
+                literal_val = int(literal.value)
+                divisor = math.ceil(literal_val / step_val)
+                modulo = literal_val % step_val
+                # If the divisor is > 1, then we need to do
+                # divisor*step_val
+                # We also need to add divisor-1*step_val to cover the case
+                # where e.g. array(i+1) is inside a larger loop, as we
+                # need dependencies to array(i) and array(i+step), unless
+                # modulo == 0
+                step = None
+                step2 = None
+                if divisor > 1:
+                    step = BinaryOperation.create(
+                        BinaryOperation.Operator.MUL,
+                        Literal(f"{divisor}", INTEGER_TYPE),
+                        Literal(f"{step_val}", INTEGER_TYPE),
+                    )
+                    if divisor > 2:
+                        step2 = BinaryOperation.create(
+                            BinaryOperation.Operator.MUL,
+                            Literal(f"{divisor-1}", INTEGER_TYPE),
+                            Literal(f"{step_val}", INTEGER_TYPE),
+                        )
+                    else:
+                        step2 = Literal(f"{step_val}", INTEGER_TYPE)
+                else:
+                    step = Literal(f"{step_val}", INTEGER_TYPE)
+
+                # Create a Binary Operation of the correct format.
+                binop = None
+                binop2 = None
+                if ref_index == 0:
+                    # We have Ref OP Literal
+                    binop = BinaryOperation.create(
+                        node.operator, real_ref.copy(), step
+                    )
+                    # If modulo is non-zero then we need a second binop
+                    # dependency to handle the module, so we have two
+                    # dependency clauses, one which handles each of the
+                    # step-sized array chunks that overlaps with this access.
+                    if modulo != 0:
+                        if step2 is not None:
+                            binop2 = BinaryOperation.create(
+                                node.operator, real_ref.copy(), step2
+                            )
+                        else:
+                            binop2 = real_ref.copy()
+                else:
+                    # We have Literal OP Ref
+                    binop = BinaryOperation.create(
```

Comment:
@arporter I added these lambdas, however pycodestyle is complaining about this:
`psyir/nodes/dynamic_omp_task_directive.py:277:13: E731 do not assign a lambda expression, use a def`

From googling, it looks like the recommended way to do this would instead be:
`def first_arg() : ref.copy()`
and so on instead of lambdas - are you ok with this?

Reply:
Oh, apologies, I'd commented on this elsewhere. Yes, please go with what pylint/pycodestyle says. In fact, do we need a function at all?

Reply:
Yeah - I think actually each of the options is only used once, so just having `first_arg = ref.copy()` is sufficient.

Reply:
They're all now copies though to avoid issue when creating the second binary issue of not having separate instances for each.